src/erlfmt_cli.erl

```diff
@@ -48,19 +50,34 @@ opts() ->
         {files, undefined, undefined, string, "files to format, - for stdin"}
     ].
 
--spec do(list(), string()) -> ok.
-do(Opts, Name) ->
+-spec do(string(), list()) -> ok.
+do(Name, Opts) ->
+    do(Name, Opts, []).
+
+-spec do(string(), list(), list()) -> ok.
+do(Name, PreferOpts, DefaultOpts) ->
+    PreferParsed = parse_opts(PreferOpts),
+    DefaultParsed = parse_opts(DefaultOpts),
+    Parsed = resolve_parsed(PreferParsed, DefaultParsed),
+    with_parsed(Name, Parsed).
+
+-spec with_parsed(string(), parsed()) -> ok.
+with_parsed(Name, Config) ->
     try
-        do_unprotected(Opts, Name)
+        unprotected_with_config(Name, Config)
```

Comment:
Before parsing was also protected inside the try, I'm not sure if it's important, but perhaps we should keep it. What do you think?

Reply:
parse_opts is now a pure and total function, so I hope we can get away without the catch and rather fix the bugs when we encounter an unexpected case.

Reply:
I added an extra manual test for this case